### PR TITLE
Fix use of non-null assertion operator

### LINKED ISSUE #102: Non-null assertion causes linters to fail
One of the uncompleted tasks in #71, the non-null assertion in the various parts of the typescript source code causes the GitHub linter to fail. It can be resolved by running `npm run prettier:fix`, but it appears to be a quick fix and not a long term solution.

This was discussed in https://github.com/kevinddchen/machikoro/issues/97#issuecomment-1524826316.

### PR DESCRIPTION
**Pull request by AquaDragon**
This PR fixes all but one use of the non-null assertion operator ("!") which triggers the forbidden non-null assertion error by ESLint (i.e. `@typescript-eslint/no-non-null-assertion`). It attempts to resolve #102. This PR will update the game version to v0.6.3.

Changes:
- Utilities functions were added which asserts that the value is not null and returns an error message otherwise. 
- Disabled the `@typescript-eslint/no-unused-vars` error for the assertUnreachable function. The empty variable appears unused but the function checks for the `never` type when it is input into the function.

Unresolved:
- https://github.com/kevinddchen/machikoro/blob/029872ce67b3ce03c17b2c67ad82c0a7013294a7/src/game/machikoro.ts#L1151
`Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion`
Not sure how to assert a type here. `PlayerView` was imported and the type seems complicated.

- https://github.com/kevinddchen/machikoro/blob/029872ce67b3ce03c17b2c67ad82c0a7013294a7/src/server/middleware.ts#L72
`Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any`
This script was removed during a recent rollback (#105), not sure what the type should be.

### CODE REVIEW

**Comment by AquaDragon** [comment]
by the way, I used ChatGPT to help with the coding so while it works, we have to make sure its not doing something else unintended

File: src/common/typescript.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (kevinddchen) [inline src/common/typescript.ts:1]: :+1:

File: src/board/utils.tsx [hunk 2]
```diff
@@ -3,6 +3,15 @@ import { assertUnreachable } from 'common/typescript';
 
 import classNames from 'classnames';
 
+/**
+ * Function to assert type is not null to avoid using the non-null assertion operator "!".
+ */
+export function assertNonNull<T>(value: T | null | undefined): asserts value is T {
+  if (value == null) {
```
Comment (kevinddchen) [inline src/board/utils.tsx:10]: maybe `if (value === null || value === undefined)` is better here

Comment (kevinddchen) [reply src/board/utils.tsx:10]: also, merge all 4 (?) instances of this function into a single function at "src/common/typescript.ts"

**Review by kevinddchen — APPROVED** [review]
lgtm!

Couple suggestions:
- Merge all instances of `assertNonNull` into a single instance. In general, its good practice to eliminate copied code whenever you can.
- `PlayerView.STRIP_SECRETS` can be ignored via comment. We have 0 control over that.
- Delete the "src/server/middleware.ts" file. There are no code errors if there is no code!

**Comment by AquaDragon** [comment]
import { assertNonNull, assertUnreachable } from 'common/typescript';

\machikoro\src\lobby\Lobby.tsx
20:1  warning  Imports should be sorted alphabetically  sort-imports


are you serious, eslint?

**Pull request merged**